src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -900,14 +901,100 @@ private static bool SupportsFoldingRange(JToken token)
                 },
             };
 
-            var textBuffer = virtualDocumentSnapshot.Snapshot.TextBuffer;
-            var response = await _requestInvoker.ReinvokeRequestOnServerAsync<CompletionParams, JToken?>(
-                textBuffer,
-                Methods.TextDocumentCompletion.Name,
-                languageServerName,
-                completionParams,
-                cancellationToken).ConfigureAwait(false);
-            return response?.Response;
+            var continueOnCapturedContext = false;
+            var provisionalTextEdit = request.ProvisionalTextEdit;
+            if (provisionalTextEdit is not null)
+            {
+                await _joinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+                var provisionalChange = new VisualStudioTextChange(request.ProvisionalTextEdit, virtualDocumentSnapshot.Snapshot);
+                UpdateVirtualDocument(provisionalChange, request.ProjectedKind, request.HostDocument.Version, documentSnapshot.Uri);
+
+                // We want the delegation to continue on the captured context because we're currently on the `main` thread and we need to get back to the
+                // main thread in order to update the virtual buffer with the reverted text edit.
+                continueOnCapturedContext = true;
```

Comment:
~This works?! The `ConfigureAwait(true)` is immediately before the end of the method, so i would have thought the thread that the calling method ends up on would be up to how it called this method, and there is nothing this method could do to influence it.~

EDIT: durr its the finally. nevermind